Dockerfile

```diff
@@ -1,5 +1,14 @@
+ARG GID=1000
+ARG UID=1000
+
 # Build the manager binary
 FROM golang:1.18 as builder
+ARG GID
+ARG UID
+
+# Create user and group
+RUN groupadd -g ${GID} appgroup && \
```

Comment:
do you need to do it for the builder too?  maybe the copies doesn't work from builder if you don't make them both non-root?

Reply:
I might not have understood the question correctly. The docker build is done in 2 stages `builder` and `base`. The `builder` used goland docker container to build binary and create `appuser:appgroup` which creates file `/etc/passwd` & `/etc/group`.
These two files and binary is copied to base image `distroless` so that base image runs binary with `appuser` as user

Reply:
`ARG GID and ARG UID` is re-declared to passed values from above definition to both of these two build stages